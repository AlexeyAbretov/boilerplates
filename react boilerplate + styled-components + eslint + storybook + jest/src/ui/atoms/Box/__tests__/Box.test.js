import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import {
  Box,
} from '..';

describe('ui/atoms/Box', () => {
  test('Box should be rendered', () => {
    const component = renderer.create(
      <Box>It&apos;s the Box</Box>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Box should be rendered with border', () => {
    const component = renderer.create(
      <Box border="1px solid black">It&apos;s the Box</Box>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Box test click', () => {
    const wrapper = shallow(
      <Box onClick={() => alert('clicked')}>It&apos;s the Box</Box>,
    );

    wrapper.simulate('click');
    expect(global.alert).toHaveBeenCalledWith('clicked');
  });
});
