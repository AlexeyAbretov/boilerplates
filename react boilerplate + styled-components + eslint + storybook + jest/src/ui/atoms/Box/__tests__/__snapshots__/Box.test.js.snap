// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/atoms/Box Box should be rendered 1`] = `
<div
  className="Box-cYgLTd"
>
  It's the Box
</div>
`;

exports[`ui/atoms/Box Box should be rendered with border 1`] = `
.c0 {
  border: 1px solid black;
}

<div
  className="c0"
>
  It's the Box
</div>
`;
